packages/clerk-js/src/ui/contexts/utils.ts

```diff
@@ -13,3 +13,16 @@ export function assertContextExists(contextVal: unknown, providerName: string):
     clerkCoreErrorContextProviderNotFound(providerName);
   }
 }
+
+export function isRedirectForFAPIInitiatedFlow(clerk: Clerk, redirectUrl: string): boolean {
+  const url = new URL(redirectUrl);
+  const path = url.pathname;
+
+  return clerk.frontendApi === url.host && frontendApiRedirectPaths.includes(path);
+}
+
+const frontendApiRedirectPaths: string[] = [
+  '/oauth/authorize', // OAuth2 identify provider flow
+  '/v1/verify', // magic links
+  '/v1/tickets/accept', // ticket flow
+];
```

Comment:
🙃 for object constants prefer using `const variable = [...] as const` to avoid mistakenly mutating them.